ui/app/contexts/metametrics.new.js

```diff
@@ -0,0 +1,180 @@
+      if (match) {
+        const { path, params, url } = match
+        const name = PATH_NAME_MAP[path]
+        segment.page({
+          [idTrait]: idValue,
+          name,
+          properties: {
+            url,
+            hash: location.hash,
+            // We do not want to send addresses or accounts in any events
+            // Some routes include these as params.
+            params: omit(params, ['account', 'address']),
+          },
+        })
+      } else if (location.pathname !== '/confirm-transaction') {
+        // We have more specific pages for each type of transaction confirmation
+        // The user lands on /confirm-transaction first, then is redirected based on
+        // the contents of state.
+        captureMessage(`${location.pathname} would have issued a page track event to segment, but no route match was found`)
+      }
+    }
+  }, [location, metaMetricsId, participateInMetaMetrics])
+
+  /**
+   * track a metametrics event using segment
+   * e.g metricsEvent({ event: 'Unlocked MetaMask', category: 'Navigation' })
+   *
+   * @param {object}   config - configuration object for the event to track
+   * @param {string}   config.event - event name to track
+   * @param {string}   config.category - category to associate event to
+   * @param {boolean}  [config.isOptIn] - happened during opt in/out workflow
+   * @param {object}   [config.properties] - object of custom values to track, snake_case
+   * @return {undefined}
+   */
+  const trackEvent = useCallback(
+    (config = {}) => {
+      console.log(config)
+      const { event, category, isOptIn = false, properties = {} } = config
+      if (!event) {
+        // Event name is required for tracking an event
+        throw new Error('MetaMetrics trackEvent function must be provided a payload with an "event" key')
+      }
+      if (!category) {
+        // Category must be supplied for every tracking event
+        throw new Error('MetaMetrics events must be provided a category')
+      }
+
+      let excludeMetaMetricsId = config.excludeMetaMetricsId ?? false
+
+      // This is carried over from the old implementation, and will likely need
+      // to be updated to work with the new tracking plan. I think we should use
+      // a config setting for this instead of trying to match the event name
+      const isSendFlow = Boolean(event.match(/^send|^confirm/u))
+      if (isSendFlow && !sendCountIsTrackable(metaMetricsSendCount + 1)) {
+        excludeMetaMetricsId = true
+      }
+      const idTrait = excludeMetaMetricsId ? 'anonymousId' : 'userId'
+      const idValue = excludeMetaMetricsId ? METAMETRICS_ANONYMOUS_ID : metaMetricsId
+
+      if (participateInMetaMetrics || isOptIn) {
+        segment.track({
+          [idTrait]: idValue,
+          event,
+          properties: {
+            ...properties,
+            category,
+            exclude_meta_metrics_id: excludeMetaMetricsId,
+          },
+          context: {
```

Comment:
I'm not sure this will work :thinking: 

https://segment.com/docs/connections/spec/common/#context

>Context is a complete and explicit specification, so properties outside the spec are ignored. You should only use Context fields for their intended meaning.

Reply:
Perhaps these could just always be added to `properties`? At least for those without an equivalent `context` property.

Reply:
hmm i'll test it out and see if they end up in segment.

Reply:
![image](https://user-images.githubusercontent.com/4448075/92816269-dacb1480-f38a-11ea-9d0c-1246d97c62ad.png) they come through segment but then don't make it into mixpanel.

Reply:
@Gudahtt -- I think I can put version in app, so 

```
app: {
  version: platform.getVersion(),
  build: process.env.METAMASK_ENVIRONMENT,
},
```

network maybe goes into an identify call or attached as a property to every event? environment_type won't be able to go anywhere. I added userAgent which correctly identifies the browser in mixpanel so we can avoid that one.

Reply:
Putting those two things in `app` sounds good! Though for `build` it'd be nice to use the git hash instead :thinking: I had been meaning to add that to Sentry as well, and the git hash would make for a better build number. Many verison + environment combinations would have multiple builds.

Maybe it would be more appropriate to put the environment in the version? e.g. something like

```
let version = platform.getVersion()
if (process.env.METAMASK_ENVIRONMENT !== 'production') {
  version = `${version}-${process.env.METAMASK_ENVIRONMENT}`
}
```

Reply:
`network` doesn't seem like a great candidate for identify, since it can change from minute-to-minute. I would go the property route. `environment_type` seems like it could be a property as well :thinking:

Reply:
done in https://github.com/MetaMask/metamask-extension/pull/9382/commits/f5f1c3d7d24e1cd4fcd68b1ee0dbe06017a949d9